cni/network/invoker_cns_test.go

```diff
@@ -2094,3 +2094,34 @@ func TestCNSIPAMInvoker_Add_SwiftV2(t *testing.T) {
 		})
 	}
 }
+
+func TestShallowCopyIpamAddConfigOptions(t *testing.T) {
+	opts := IPAMAddConfig{
+		// mock different types of map value
+		options: map[string]interface{}{
+			network.SNATIPKey:   "10",
+			dockerNetworkOption: "20",
+			"intType":           10,
+			"floatType":         0.51,
+			"byteType":          byte('A'),
+		},
+	}
+
+	// shallow copy all ipamAddConfig options
+	res := opts.shallowCopyIpamAddConfigOptions()
+	require.Equal(t, opts.options, res)
+
+	// create a second opts for reference representing expected outputs
+	copyRes := res
+	// modify res1 fields and make sure copied res has changed
```

Comment:
This doesn't copy res. I believe it just copies the pointer to the underlying map. While this is valid, we want to test that after a shallow copy and modification of the copy, the original _has not_ changed.